src/ScriptBuilder/CSharpCounterWriter.cs

```diff
@@ -13,23 +13,28 @@ public static void WriteCode(string scriptPath, IEnumerable<DurationDefinition>
             {
                 var stringBuilder = new StringBuilder();
 
-                var slaCounterDefinition = @"new CounterCreationData(""SLA violation countdown"", ""Seconds until the SLA for this endpoint is breached."", PerformanceCounterType.NumberOfItems32),";
+                var slaCounterDefinition = $@"new CounterCreationData(""SLA violation countdown"", ""{CounterNameAndDescriptionConventions.LegacySlaViolationCounterDescription}"", PerformanceCounterType.NumberOfItems32),";
                 stringBuilder.AppendLine(slaCounterDefinition.PadLeft(slaCounterDefinition.Length + 8));
 
                 foreach (var duration in durations)
                 {
                     var averageTimerName = duration.Name.GetAverageTimerCounterName();
                     var averageTimerBase = duration.Name.GetAverageTimerBaseCounterName();
+                    var averateTimerDescription = CounterNameAndDescriptionConventions.GetAverageTimerDescription(duration.Name, duration.Description);
 
-                    var durationAverageDefinition = $@"new CounterCreationData(""{averageTimerName}"", ""{duration.Description}"", PerformanceCounterType.AverageTimer32),";
+                    var durationAverageDefinition = $@"new CounterCreationData(""{averageTimerName}"", ""{averateTimerDescription}"", PerformanceCounterType.AverageTimer32),";
```

Comment:
Ok, so that means the performance counters api does use the whole long type value in its calculation. Why do we have a `AverageTimer32` then :-). Then this comment can be discarded.